site/cds_rdm/tasks.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2023 CERN.
+#
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Celery tasks for cds."""
+import requests
+from celery import shared_task
+from flask import current_app
+from invenio_oauthclient.handlers.base import create_or_update_groups
+
+
+@shared_task
+def sync_groups():
+    """Synchronizes groups in CDS."""
+    token_url = (
+        f"{current_app.config['CERN_KEYCLOAK_URL']}auth/realms/cern/api-access/token"
+    )
+    token_data = {
+        "grant_type": "client_credentials",
+        "client_id": current_app.config["CERN_APP_CREDENTIALS"]["consumer_key"],
+        "client_secret": current_app.config["CERN_APP_CREDENTIALS"]["consumer_secret"],
+        "audience": "authorization-service-api",  # This is the target api of the token
+    }
+
+    toke_response = requests.post(url=token_url, data=token_data)
+    token = toke_response.json()["access_token"]
+    offset = 0
+    limit = 1000
+    groups_headers = {
+        "Authorization": f"Bearer {token}",
+        "accept": "text/plain",
+    }
+
+    # We do this to get the total amount of entries, to be able to create as many celery tasks as required
+    groups_url = f"{current_app.config['AUTHORIZATION_SERVICE_API']}api/v1.0/Group?offset=0&limit=1"
+    groups_response = requests.get(url=groups_url, headers=groups_headers).json()
+    total = groups_response["pagination"]["total"]
+    while offset < total:
+        update_groups.delay(offset, limit, groups_headers)
```

Comment:
moderate: in a scenario where we need to update 10 batches (i.e. 10k groups). Imagine 4 updates go well, but the other 6 fail (e.g. network errors, etc.). Then the update is not atomic, it stays inconsistent. Is this acceptable, thinking that the next update will fix it?

Reply:
For the moment, I think is good enough. If there is a error on one of the batches it will retry every 10 minutes for 1 hour, if it ends up failing it will raise an Exception, so that the sys admins are notified and can take action if needed.